### DISCUSSION TITLE
Accessories of Multiple Destinations on the same Machine

### DISCUSSION THREAD

**Discussion by JensRavens** [basecamp/kamal#1153] [Q&A]
Hello everybody,

I'm trying to create a deployment where multiple environments of the same application including their accessories are deployed to the same machine (think production, staging and automatically created apps per branch).

The first issue I run into is that accessory names don't seem to take destination names into account (so my production postgres and my staging postgres become the same container). I fixed that by using different names for accessories in different destinations (which feels wrong because now I have to redeclare all accessories for every environment).

My next issue is how to access the accessory without explicitly exposing their port to localhost. Due to the (theoretically unlimited due to branches) number of accessories I can't just assign a port to them (especially if I reuse the server to deploy multiple applications with their various destinations as I would run out of port numbers at some point). My hope was to access the accessories via their docker names (which is what I used to do in a similar setup with docker compose) like this:

- app_a
  - production
    - production-postgres
  - staging
    - staging-postgres
- app_b
  - production
    - production-postgres

Then set the `DATABASE_URL` via env to `postgres://postgres@app_a-production-postgres:5432/dbname`, but the host name cannot be resolved by my application. 

**So to summarise:**
- Is it a bug that accessories for multiple destinations are shared? Should I work around that or open a bug report instead?
- Can I access an accessory without exposing a port? E.g. with a custom docker network or something similar?

**Comment by rorykoehler** [comment]
Did you resolve this?

**Reply by JensRavens** [reply]
Sadly no. I’m thinking about somehow automatically assigning ports and keeping a log with a rake task, but that sounds overcomplicated.

**Comment by koenhandekyn** [comment]
i'm doing something similar (dev/staging/prod) but it's much simpler to share the pg database, as pg as schemas that provide full separation, similar to what you have on heroku also for example. running seperate pg's
